CMakeLists.txt

```diff
@@ -25,6 +25,9 @@ set(PROJECT_VERSION "2.3.0")
 # Avoid SYCL compiler error
 if(NOT WIN32)
   string(APPEND CMAKE_CXX_FLAGS " -Wno-error")
+  if("$ENV{IS_XPU_CI}" STREQUAL "1")
+    string(APPEND CMAKE_CXX_FLAGS " -Werror=unused-variable")
```

Comment:
Should we always turn on this build option?

Reply:
Personally, I do not recommend always turning on this build option for the reason I mentioned.

Reply:
I got your point @CuiYifeng - let's just keep this flag for CI builds